pytorch_lightning/utilities/enums.py

```diff
@@ -49,6 +49,29 @@ class AMPType(LightningEnum):
     NATIVE = "native"
 
 
+class PrecisionType(LightningEnum):
+    """Type of precision used.
+
+    >>> PrecisionType.HALF == 16
+    True
+    >>> PrecisionType.HALF in (16, "16")
+    True
+    """
+
+    HALF = 16
+    FLOAT = 32
+    FULL = 64
+    BFLOAT = "bf16"
+
+    @staticmethod
+    def supported_type(precision: Union[str, int]) -> bool:
+        return any(x == precision for x in PrecisionType)
+
+    @staticmethod
+    def supported_types() -> List[str]:
```

Comment:
```suggestion
    def supported_types() -> List[Union[int, str]]:
```

Reply:
This is never true, because I think the enum will always return a string